lib/utils/templatize.js

```diff
@@ -375,20 +379,32 @@ function createTemplatizerClass(template, templateInfo, options) {
  * 
  * @suppress {missingProperties} class.prototype is not defined for some reason
  */
-function addPropagateEffects(template, templateInfo, options, methodHost) {
+function addPropagateEffects(target, templateInfo, options, methodHost) {
   let userForwardHostProp = options.forwardHostProp;
   if (userForwardHostProp && templateInfo.hasHostProps) {
+    // Under the `removeNestedTemplates` optimization, a custom element like
+    // `dom-if` or `dom-repeat` can itself be treated as the "template"; this
+    // flag is used to switch between upgrading a `<template>` to be a property
+    // effects client vs. adding the effects directly to the custom element
+    const isTemplate = target.localName == 'template';
     // Provide data API and property effects on memoized template class
     let klass = templateInfo.templatizeTemplateClass;
     if (!klass) {
       /**
        * @constructor
        * @extends {DataTemplate}
        */
-      let templatizedBase = options.mutableData ? MutableDataTemplate : DataTemplate;
-      /** @private */
-      klass = templateInfo.templatizeTemplateClass =
-        class TemplatizedTemplate extends templatizedBase {};
+      if (isTemplate) {
+        let templatizedBase = options.mutableData ? MutableDataTemplate : DataTemplate;
+        /** @private */
+        klass = templateInfo.templatizeTemplateClass =
+          class TemplatizedTemplate extends templatizedBase {};
+      } else {
+        // Create a cached subclass of the base custom element class onto which
```

Comment:
How is this cached?

Reply:
Could this also be `Object.create(Object.getPrototypeOf(target)`? It's not just because we're otherwise expecting a class here?

Reply:
Fix the caching by setting `templateInfo.templatizeTemplateClass = klass`

Reply:
Done